### PR TITLE
Feat/pupil 892/quiz result question stem

### PR DESCRIPTION
**Pull request by benprotheroe**
## Description

Music year: 1976

- List of changes
created QuizResultsQuestionStem component and story

and added a story to quizquestionstem (just because)

## Issue(s)

Fixes #

## How to test

1. Go to https://deploy-preview-2651--owa-storybook.netlify.thenational.academy/?path=/story/components-pupilcomponents-quizresultquestionstem--default
2. Click on default/ text and image 
3. You should see the question stem variations

## Screenshots

How it used to look (delete if n/a):
{screenshots}

How it should now look:
{screenshots}

## Checklist

- [ ] Added or updated tests where appropriate
- [ ] Manually tested across browsers / devices
- [ ] Considered impact on accessibility
- [ ] Design sign-off
- [ ] Approved by product owner
- [ ] Does this PR update a package with a breaking change

### CODE REVIEW

**Review by messmike — APPROVED** [review]

**Comment by RBrownsett** [comment]
@benprotheroe looks good - storybook is throwing an accessibility issue to do with the role of presentation being used - I can't see it in the code, only used in your test. Any ideas? 

image

image

**Comment by RBrownsett** [comment]
> > Details
> 
> Screenshot 2024-07-31 at 15 26 44
> its coming from this spriteSheet component; not sure why its appearing on the story; but when its used on owa its the object tag isnt there https://www.thenational.academy/pupils/programmes/maths-secondary-year-11-higher/units/real-life-graphs/lessons/checking-and-securing-understanding-of-reading-from-context-based-graphs/exit-quiz
> 
> @RBrownsett

Ah so it is - how odd that it turned up in the storybook component. Thanks for looking into it

**Review by RBrownsett — APPROVED** [review]
Approved - tested, no issues found

File: src/components/PupilComponents/QuizQuestionStem/QuizQuestionStem.tsx [hunk 1]
```diff
@@ -4,25 +4,16 @@ import {
   OakSpan,
 } from "@oaknational/oak-components";
 
+import { QuizQuestionStemProps } from "../QuizResultQuestionStem/QuizResultQuestionStem";
```
Comment (kimon-satan) [inline src/components/PupilComponents/QuizQuestionStem/QuizQuestionStem.tsx:7]: I 'm guessing sonar cloud made you do this, but it's not good. The props should not be shared between these two components.

File: src/components/PupilComponents/QuizResultQuestionStem/QuizResultQuestionStem.tsx [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+      $flexDirection={"column"}
+      $gap={"space-between-m"}
+      $color={"text-primary"}
+      $height={takeFullHeight ? "100%" : "auto"}
+      $justifyContent={["center", "flex-start"]}
+    >
+      <OakFlex
+        key="stem-header"
+        $mt={
+          takeFullHeight
```
Comment (kimon-satan) [inline src/components/PupilComponents/QuizResultQuestionStem/QuizResultQuestionStem.tsx:38]: We shouldn't need this margin in this version of the question stem ?

File: src/components/PupilComponents/QuizResultQuestionStem/QuizResultQuestionStem.tsx [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+  removeMarkdown,
+  shortAnswerTitleFormatter,
+} from "@/components/TeacherComponents/LessonOverviewQuizContainer/quizUtils";
+import {
+  ImageItem,
+  TextItem,
+} from "@/node-lib/curriculum-api-2023/queries/pupilLesson/pupilLesson.schema";
+import { getSizes } from "@/components/SharedComponents/CMSImage/getSizes";
+
+export type QuizQuestionStemProps = {
```
Comment (kimon-satan) [inline src/components/PupilComponents/QuizResultQuestionStem/QuizResultQuestionStem.tsx:17]: If we're going to have a shared type for props between two components then they need to be stored in a higher order folder. But actually I don't think it's appropriate here.

File: src/components/PupilComponents/QuizResultQuestionStem/QuizResultQuestionStem.tsx [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+            $font={"body-2-bold"}
+          >
+            {`${displayNumber} ${shortAnswerTitleFormatter(
+              removeMarkdown(questionStem[0].text),
+            )}`}
+          </OakSpan>
+        )}
+      </OakFlex>
+
+      {questionStem.map((stemItem, i) => {
```
Comment (kimon-satan) [inline src/components/PupilComponents/QuizResultQuestionStem/QuizResultQuestionStem.tsx:55]: If SonarCloud is still complaining then this is probably the portion that could be refactored to a sub component used by both.

**Review by kimon-satan — CHANGES_REQUESTED** [review]

File: src/components/PupilComponents/QuizQuestionStem/QuizQuestionStem.tsx (hunk 1 above)
Comment (benprotheroe) [reply src/components/PupilComponents/QuizQuestionStem/QuizQuestionStem.tsx:7]: solution?

File: src/components/PupilComponents/QuizResultQuestionStem/QuizResultQuestionStem.tsx (hunk 3 above)
Comment (benprotheroe) [reply src/components/PupilComponents/QuizResultQuestionStem/QuizResultQuestionStem.tsx:17]: ok..

File: src/components/PupilComponents/QuizResultQuestionStem/QuizResultQuestionStem.tsx (hunk 4 above)
Comment (benprotheroe) [reply src/components/PupilComponents/QuizResultQuestionStem/QuizResultQuestionStem.tsx:55]: its not

Comment (kimon-satan) [reply src/components/PupilComponents/QuizResultQuestionStem/QuizResultQuestionStem.tsx:55]: I meant after you've made my suggested change

File: src/components/PupilComponents/QuizQuestionStem/QuizQuestionStem.tsx (hunk 1 above)
Comment (kimon-satan) [reply src/components/PupilComponents/QuizQuestionStem/QuizQuestionStem.tsx:7]: I suggest it further down in the comments. Give me a call if unsure

**Review by kimon-satan — APPROVED** [review]

**Pull request merged**